atecc608a_se.c

```diff
@@ -418,6 +418,9 @@ static psa_status_t atecc608a_allocate_key(
     (void) method;
     (void) key_slot;
 
+    /* TODO - this function is not implemented yet. It will contain
+       Mbed OS persistent memory handling to ensure that a single slot
```

Comment:
No, not "Mbed OS persistent memory handling". Drivers should not modify external storage during allocate, even on a platform where they can. That would make recovery very difficult if there is a power failure during the key creation process. Drivers should use `persistent_data`, which exists precisely for this reason. This driver is our real-life example of how to use the `persistent_data` feature, so it's important to get it right.